#!/usr/bin/env python
PACKAGE = "mission_manager"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("waypoint_threshold", double_t, 0, "Distance from waypoint to deem reached, in m",    10.0, 1.0, 25.0)

gen.add("turn_radius",        double_t, 0, "Minimum turn radius, in m",                       20.0, 1.0, 100.0)
gen.add("segment_length",     double_t, 0, "Length of segment when decomposing curves, in m",  5.0, 1.0, 100.0)

gen.add("default_speed",      double_t, 0, "Default speed to use in missions, in m/s",         3.0, 0.0, 10.0)

planner_enum = gen.enum([ gen.const("PathFollower", int_t, 0, "Path Follower"),
                          gen.const("PathPlanner",  int_t, 1, "Path Planner")],
                        "Planner to use to follow a path.")
                        
gen.add("planner", int_t, 0, "Planner to use", 0, 0, 1, edit_method=planner_enum)

exit(gen.generate(PACKAGE, "mission_manager", "mission_manager"))
